daml-lf/validation/src/test/scala/com/digitalasset/daml/lf/validation/SerializabilitySpec.scala

```diff
@@ -220,6 +220,59 @@ class SerializabilitySpec extends WordSpec with TableDrivenPropertyChecks with M
       }
 
     }
+
+    "reject unserializable contract id" in {
+
+      val pkg =
+        p"""
+          // well-formed module
+          module NegativeTestCase1 {
+            record @serializable SerializableRecord = {};
+
+            template (this : SerializableRecord) =  {
+              precondition True,
+              signatories Nil @Party,
+              observers Nil @Party,
+              agreement "Agreement",
+              choices {
+              }
+            } ;
+
+            record @serializable SerializableContractId = { cid : ContractId NegativeTestCase1:SerializableRecord };
+          }
+
+          module NegativeTestCase2 {
+            record @serializable SerializableContractId = { cid : ContractId NegativeTestCase1:SerializableRecord };
+          }
+
+          module PositiveTestCase1 {
+            record @serializable SerializableRecord = {};
+
+            record @serializable UnserializableContractId = { cid : ContractId PositiveTestCase1:SerializableRecord };
```

Comment:
@remyhaemmerle-da This test fails because an `EUnknownDefinition` exception is thrown. If I replace `PositiveTestCase1:SerializableRecord` with just `SerializableRecord` then the test passed. If I replace it with `SerializableRecordXYZ`, it still passes even though it should fail with `EUnknownDefinition` definition. This doesn't make any sense. Could you please have a look.